rfc-146-production-deploy-access.md

```diff
@@ -0,0 +1,82 @@
+
+Note that by this logic, the 2ndline requirements are defending against accidental mistakes, not against malicious access. We have separate defences against malicious employees (security clearance, probation etc.).
+
+## Proposal
+
+We should implement a new level of access - "Production Deploy Access". This level of access should include:
+
+- Permission to deploy apps in Jenkins (ideally without Jenkins admin permission)
+- Permission to merge continuously deployed applications
+- Readonly access to AWS, Fastly, logging systems, etc.
+- admin access to GOV.UK Signon (create and edit normal users)
+
+This level of access should not include:
+
+- ssh permission to production or staging
+- admin access to the Deploy Jenkins in production or staging
+- admin or poweruser access to AWS
+- admin access to other systems (e.g. Fastly etc.)
+- superadmin access to GOV.UK Signon (create and edit all user types and edit applications)
+
+Engineers should be granted this level of access once they've got the required level of security clearance and have been working on GOV.UK for at least one full month. Access should be granted at the discretion of the engineer's tech lead. Before approving access, tech leads should ensure that the engineer:
+
+- is aware of our processes and standards around code review
+- understands the responsibilities that releasing code brings with it
+- knows how to roll back to an older release if there are any issues
+- knows how to get help from someone with more access if they need it
+
+These rules should be the same for contractors and civil servants at all levels. Junior and apprentice developers should be given production deploy access once their tech lead feels it's appropriate.
+
+The rules for gaining Production Admin Access should remain the same - engineers will still be required to complete two 2ndline shifts.
+
+## Consequences
+
+We will update the [Rules for Getting Production Access](https://docs.publishing.service.gov.uk/manual/rules-for-getting-production-access.html) to match the outcome of this RFC.
+
+We will create a new GitHub Team for the middle level of access, so we'll have "GOV.UK", "GOV.UK Deploy", and "GOV.UK Production".
+
+We will set up the staging and production deploy Jenkins instances to allow people in the GOV.UK Deploy GitHub Team to trigger builds.
+
+The bottleneck of needing people with production access to merge and deploy code will be substantially widened, since it will be much easier to give people deploy access.
+
+Teams may be less willing to have new engineers doing 2ndline shifts, since the carrot of production access will be a bit less appetising if people can get deploy access without the shifts. We'll need to keep an eye on this, since our current position is that 2ndline shifts are independently valuable things for engineers to do.  
```

Comment:
If this theory were correct we'd be seeing a drop off of engineer availability after they'd done the minimum for access. Are we? I think that typically teams are reluctant to have engineers do 2nd line shifts because it's disruptive but reluctantly support the model because there's no real alternative.

Reply:
Yeah, I don't think we do see that, so perhaps this is an unhelpful hypothetical. I'd be happy to remove this paragraph (but let's leave it for a few days incase anyone else has thoughts on this).

Reply:
Agree this paragraph should be removed. Our 2nd line slots aren't volunteer-only, they're mandatory and allocated, so how a team feels about doing shifts shouldn't be relevant.